packages/jsapi-types/src/dh.types.ts

```diff
@@ -335,6 +336,17 @@ export interface Figure extends Evented {
   close(): void;
 }
 
+export interface Widget {
+  readonly EVENT_MESSAGE: string;
+
+  addEventListener: (
+    type: string,
+    listener: (event: unknown) => void
```

Comment:
Might want to make a separate interface for the message payload and also we should add another signature for the message event. There is also a close event which I believe is only fired on server sending a not ok status. We probably need another status/event to differentiate server hanging up vs server disconnecting due to something like network instability.

```
addEventListener: (
  type: 'message', // Or typeof EVENT_MESSAGE and set that to its value
  listener: (event: CustomEvent<Omit<Widget, 'addEventListener', 'sendMessage', 'EVENT_MESSAGE'>>) => void
)
```

Reply:
Yea, ultimately it extends the `HasEventHandling` interface... I don't really want to do too much here right now since I want to publish the automatic types and then use those, would like to see how that shakes out before re-writing this.